.NET/Microsoft.Recognizers.Text.DateTime/Parsers/CJK/BaseCJKDatePeriodParser.cs

```diff
@@ -837,10 +873,19 @@ private DateTimeResolutionResult ParseDayToDay(string text, DateObject reference
                     var endDateForPastResolution = DateObject.MinValue.SafeCreateFromValue(endYearForPastResolution, endMonthForPastResolution, endDay);
                     var beginDateForFutureResolution = DateObject.MinValue.SafeCreateFromValue(beginYearForFutureResolution, beginMonthForFutureResolution, beginDay);
                     var endDateForFutureResolution = DateObject.MinValue.SafeCreateFromValue(endYearForFutureResolution, endMonthForFutureResolution, endDay);
-                    var beginTimex = DateTimeFormatUtil.LuisDate(undefinedValue, undefinedValue, beginDay);
-                    var endTimex = DateTimeFormatUtil.LuisDate(undefinedValue, undefinedValue, endDay);
 
-                    ret.Timex = $"({beginTimex},{endTimex},P{durationDays}D)";
+                    if (relativeMonth.Success)
+                    {
+                        beginLuisStr = DateTimeFormatUtil.LuisDate(beginYearForFutureResolution, beginMonthForFutureResolution, beginDay);
+                        endLuisStr = DateTimeFormatUtil.LuisDate(endYearForFutureResolution, endMonthForFutureResolution, endDay);
+                    }
+                    else
+                    {
+                        beginLuisStr = DateTimeFormatUtil.LuisDate(undefinedValue, undefinedValue, beginDay);
+                        endLuisStr = DateTimeFormatUtil.LuisDate(undefinedValue, undefinedValue, endDay);
+                    }
+
+                    ret.Timex = $"({beginLuisStr},{endLuisStr},P{durationDays}D)";
```

Comment:
Please declare a method in timexutility that receives the correct parameters and generates the timex. The whole date to string part should probably also happen there.

Reply:
replaced with GenerateDatePeriodTimex